enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,450 @@
+parts of the imagestreams and imageregistry functionality which would need to be changed to support manifestlisted images.
+
+### Samples
+
+As part of the changes to the Imagestreams API, the ImagestreamImport API's spec field would include a [new flag][IR-214] which would import the manifestlist
+rather than importing only an image from the manifestlist. The samples imagestreams would be updated to set this new flag. Not all samples will support
+all architectures, so in the future imagestreams would have to be annotated to specify which architectures they support. This new flag would be turned "on"
+by default importing manifestlists if the image is manifestlisted.
+
+### OLM
+
+OLM would need to make sure that it deploys operators only to nodes whose architecture matches the architectures on which the operator is supported.
+Today, OLM has a subscription object which is used to tell it how to deploy an operator. It has support for the nodeselector field but this would
+not be sufficient to specify filtering on multiple architectures. To this end, it would have to support the nodeAffinity field which uses the
+nodeselectorterms to filter on multiple values.
+
+Instead of having OLM dictate the operator deployment on specific nodes, the individual operators themselves would specify which architectures
+they support through a `pod.spec.affinity.nodeAffinity` object which OLM will then provide the ability for the user to override if necessary.
+This information should also be added to the operands and potentially be made configurable by the user requesting the operand.
+
+Essentially, when operator authors are declaring support for multiple architectures, that entails:
+
+- Making a declaration of that support via the CSV architecture annotation
+- Setting node affinity in the CSV deployment spec so that the operator is scheduled correctly.
+- Update the operator code such that when it creates/updates operand workloads, it sets those same nodeAffinity settings that are set on
+  the operator deployment.
+
+### Machine Config Operator
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would
+extract the relevant architecture's machine-os-content based on the node it runs on. In the future, architecture specific configurations should
+not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale
+based on page size.
+
+### Machine API
+
+- As of today, machinesets can be mutated to change the instance type and image id and this could potentially result in inconsistency between existing
+machinesets and newly created machinesets. If the instance type and image is changed to one of a different architecture, there would be workers of
```

Comment:
do we mean to say "inconsistency between existing machines and newly created machines" here, not machinesets?